### PR TITLE
User can Register for an account  to view  premium content

### PR DESCRIPTION
**Pull request by jysmys**
[PT link](https://www.pivotaltracker.com/story/show/172888074)
```
As a User,
In order to access unique features on the news app,
I would like to be able to register for an account on the site.
```

- Added component for signup form
- Added route to sign up
- Added feature test for both successful path and sad path

### Snapshot
image

### CODE REVIEW

File: cypress/fixtures/successful_signup.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "data": {
+    "id": 1,
+    "role": "user",
+    "email": "user@mail.com",
+    "provider": "email",
+    "uid": "user@mail.com",
+    "allow_password_change": false
+  }
+}
```
Comment (jysmys) [inline cypress/fixtures/successful_signup.json]: Maybe uneccasary to add this fixture, but was unsure if it was the same as login so that is why....

**Review by erikbjoern — APPROVED** [review]
Looks good! For some reason the font is different than on the login page it seems like? But that can be addressed later anyway.

**Review by kermit-klein — APPROVED** [review]

File: src/components/SignUpForm.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    e.preventDefault();
+    try {
+      let response = await auth.signUp({
+        email: e.target.email.value,
+        password: e.target.password.value,
+        password_confirmation: e.target.passwordConfirmation.value,
+      });
+      if (response.data.success) {
+        props.setUid(response.data.uid);
+        history.push("/sign_in");
```
Comment (ViaMarcus) [inline src/components/SignUpForm.jsx:21]: Is this setting the uid as if we were logged in, but it is not setting authenticated after register?
Would it not be better to just set the user as authenticated on success?

File: src/components/SignUpForm.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+              type="email"
+              id="email"
+              placeholder="email"
+            ></Input>
+            <h4>Password</h4>
+            <Input
+              name="password"
+              type="password"
+              id="password"
+              placeholder="Password"
```
Comment (ViaMarcus) [inline src/components/SignUpForm.jsx]: We are not using placeholders in the login form. We should be consistent with styling across similar forms. As we have a label for the field, the placeholder isn't really necessary.

File: src/components/SignUpForm.jsx (hunk 2 above)
Comment (ViaMarcus) [reply src/components/SignUpForm.jsx:21]: What happens if you do history.pop() or history.push(history[history.length-2]), do you get redirected to the page you came from? Should we try this?

Comment (jysmys) [reply src/components/SignUpForm.jsx:21]: hm, well the user will have a uid set but authenticated will be false. Think that you do need to do the login step.

File: src/components/SignUpForm.jsx (hunk 3 above)
Comment (jysmys) [reply src/components/SignUpForm.jsx]: ok then i remove them, but i do like to have them...

**Review by designerofthing — APPROVED** [review]

**Review by PaulineBA — APPROVED** [review]

**Pull request merged**